bin/node/runtime/src/impls.rs

```diff
@@ -78,19 +78,17 @@ impl<T: Get<Perbill>> Convert<Fixed64, Fixed64> for TargetedFeeAdjustment<T> {
 		// determines if the first_term is positive
 		let positive = block_weight >= target_weight;
 		let diff_abs = block_weight.max(target_weight) - block_weight.min(target_weight);
-		// diff is within u32, safe.
-		let diff = Fixed64::from_rational(diff_abs as i64, max_weight as u64);
+		// TODO: This is not always safe! Diff can be larger than what fits into i64
+		// It is safe as long as `block_weight` is less than `max_weight`
+		let diff = Fixed128::from_rational(diff_abs as i64, NonZeroI128::new(max_weight.max(1) as i128).unwrap());
```

Comment:
@kianenigma This is the last thing breaking tests. I have the following comments in the test below:

```
			MaximumBlockWeight::get() / 2,
			MaximumBlockWeight::get(),
			// TODO these are not safe
			// Weight::max_value() / 2,
			// Weight::max_value(),

```

But ultimately, this is just not safe for the moment if the max_weight does get out of control somehow. Please advise on what should be done.